src/Form/AdminSettingsForm.php

```diff
@@ -346,30 +345,29 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
   private function setCredentialValues(Config $settings, array $values) {
     $settings->set('credential.account_name', $values['account_name']);
     $settings->set('credential.customer_site', $values['customer_site']);
-    $settings->set('credential.api_url', $values['api_url']);
-    $settings->set('credential.access_key', $values['access_key']);
-    if (!empty($values['secret_key'])) {
-      $settings->set('credential.secret_key', $values['secret_key']);
+    $settings->set('credential.assets_url', 'https://' . $this->removeProtocol($values['assets_url']));
+
+    $settings->clear('credential.api_url');
+    if (!empty($values['api_url'])) {
+      $settings->set('credential.api_url', 'https://' . $this->removeProtocol($values['api_url']));
     }
-    $settings->set('credential.js_path', $values['js_path']);
-    $settings->set('credential.assets_url', $values['assets_url']);
     $settings->clear('credential.oauth_url');
     if (!empty($values['oauth_url'])) {
-      $settings->set('credential.oauth_url', $values['oauth_url']);
+      $settings->set('credential.oauth_url', 'https://' . $this->removeProtocol($values['oauth_url']));
     }
   }
 
   /**
-   * Remove the protocal from an URL string.
+   * Remove the protocol and trailing slashes from an URL string.
    *
    * @param string $url
    *   URL.
    *
    * @return string
-   *   Same URL as input except with the 'http://' and 'https://' trimmed.
+   *   Same URL as input except with the 'http://' and 'https://' and trailing '/' trimmed.
    */
-  private function removeProtocal($url) {
-    return preg_replace('~^https?://~', '', $url);
+  private function removeProtocol($url) {
+    return rtrim(preg_replace('~^https?://~', '', $url), '/');
```

Comment:
Same as last comment - I feel 99% of the time, customers are just going to copy the exact URL from their emails and paste into the field, which is already without the protocol nor the trailing-slash. I personally feel any more fail-safe mechanisms into this front is diverting us.

At the meantime, if the customer HAS to be make a mistake, the result config would look broken:

<img width="596" alt="screen shot 2016-07-06 at 10 17 15" src="https://cloud.githubusercontent.com/assets/1379691/16621040/daa57946-4362-11e6-9b07-748aaeb198d9.png">